src/client/cli/commands/publish.js

```diff
@@ -60,8 +69,11 @@ module.exports = {
 
         const refs = []
         if (idSelector != null) {
-          const ref = getIn(obj, idSelector)
-          if (ref !== undefined) refs.push(ref)
+          let id = getIn(obj, idSelector)
+          if (idRegex != null) {
+            id = extractId(id, idRegex)
+          }
+          if (id !== undefined) refs.push(id)
```

Comment:
yeah, although that's not ideal.  We could make idSelector a required argument, and error if it doesn't match anything